src/test/java/org/seqdoop/hadoop_bam/TestCRAMOutputFormat.java

```diff
@@ -0,0 +1,156 @@
+package org.seqdoop.hadoop_bam;
+
+import hbparquet.hadoop.util.ContextUtil;
+import htsjdk.samtools.*;
+import htsjdk.samtools.cram.ref.ReferenceSource;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.*;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+import org.junit.Before;
+import org.junit.Test;
+import org.seqdoop.hadoop_bam.util.SAMOutputPreparer;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class TestCRAMOutputFormat {
+    private String input;
+    private String reference;
+    private TaskAttemptContext taskAttemptContext;
+    private JobContext jobContext;
+    private Configuration conf;
+
+    @Before
+    public void setup() throws Exception {
+        conf = new Configuration();
+        input = ClassLoader.getSystemClassLoader().getResource("test.cram").getFile();
+        reference = ClassLoader.getSystemClassLoader().getResource("auxf.fa").getFile();
+
+        conf.set("mapred.input.dir", "file://" + input);
+        conf.set(CRAMInputFormat.REFERENCE_SOURCE_PATH_PROPERTY, "file://" + reference);
+
+        taskAttemptContext = ContextUtil.newTaskAttemptContext(conf, mock(TaskAttemptID.class));
+        jobContext = ContextUtil.newJobContext(conf, taskAttemptContext.getJobID());
+    }
+
+    @Test
+    public void testCRAMRecordWriter() throws Exception {
+        final String cramOutFileName = "/tmp/cramout.cram";
```

Comment:
Can you create a temporary file that has a unique name and is deleted after the test runs (or the VM exits).